### PR TITLE
Apply [FhirStorageTestsFixtureArgumentSets] to integration tests

### PR DESCRIPTION
**Pull request by johnstairs**
Parameterizing the integration tests the way we do for E2E tests.
Also killing the in-memory data store

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj [hunk 1]
```diff
@@ -46,4 +46,8 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Features\Persistence\InMemory\" />
```
Comment (brandonpollett) [inline src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj]: Do we need this anymore?

File: test/Microsoft.Health.Fhir.Tests.Integration/AssemblyInfo.cs [hunk 2]
```diff
@@ -3,16 +3,6 @@
...
-{
-    public class InMemoryStorageTests : FhirStorageTestsBase
-    {
-        public InMemoryStorageTests()
-            : base(new InMemoryFhirDataStore())
-        {
-        }
-    }
-}
+[assembly: TestFramework(typeName: "Microsoft.Health.Extensions.Xunit.FixtureArgumentSetsXunitTestFramework", assemblyName: "Microsoft.Health.Extensions.Xunit")]
```
Comment (brandonpollett) [inline test/Microsoft.Health.Fhir.Tests.Integration/AssemblyInfo.cs]: nit: Kinda weird that GitHub sees this as a rename/move?

File: test/Microsoft.Health.Fhir.Tests.Integration/Persistence/CosmosDbFhirStorageTestsFixture.cs [hunk 3]
```diff
@@ -6,33 +6,28 @@
...
 using Microsoft.Health.Fhir.Core.Features.Persistence;
 using Microsoft.Health.Fhir.CosmosDb.Features.Storage;
 using Microsoft.Health.Fhir.CosmosDb.Features.Storage.StoredProcedures;
 using Microsoft.Health.Fhir.CosmosDb.Features.Storage.Versioning;
 using NSubstitute;
-using NonDisposingScope = Microsoft.Health.CosmosDb.Features.Storage.NonDisposingScope;
 
 namespace Microsoft.Health.Fhir.Tests.Integration.Persistence
 {
-    public class IntegrationTestCosmosFhirDataStore : IFhirDataStore, IDisposable
```
Comment (brandonpollett) [inline test/Microsoft.Health.Fhir.Tests.Integration/Persistence/CosmosDbFhirStorageTestsFixture.cs]: Isn't this still `IDisposable`?

Comment (johnstairs) [reply test/Microsoft.Health.Fhir.Tests.Integration/Persistence/CosmosDbFhirStorageTestsFixture.cs]: IScoped<T> extends IDisposable

File: test/Microsoft.Health.Fhir.Tests.Integration/AssemblyInfo.cs (hunk 2 above)
Comment (johnstairs) [reply test/Microsoft.Health.Fhir.Tests.Integration/AssemblyInfo.cs]: Not sure I can do anything about it...

File: src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj (hunk 1 above)
Comment (johnstairs) [reply src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj]: No

**Pull request merged**